akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst

```diff
@@ -645,4 +645,17 @@ Now:
     def deliver(destination: ActorPath)(deliveryIdToMessage: Long ⇒ Any): Unit
 
 The Java API remains unchanged and has simply gained the 2nd overload which allows ``ActorSelection`` to be
-passed in directly (without converting to ``ActorPath``).
\ No newline at end of file
+passed in directly (without converting to ``ActorPath``).
+
+
+Actor system shutdown
+---------------------
+``ActorSystem.shutdown``, ``ActorSystem.awaitTermination`` and ``ActorSystem.isTerminated`` has been
+deprecated in favor of ``ActorSystem.terminate`` and ``ActorSystem.whenTerminated```. Both returns a
+``Future[Terminated]`` value that will complete when the actor system has terminated.
+
+To get the same behavior as ``ActorSystem.awaitTermination`` block and wait for ``Future[Terminated]`` value
+with ``Await.result`` from the Scala standard library.
+
+    Await.result(system.whenTerminated, Duration.Inf)
```

Comment:
`terminate` also returns a future now right? Maybe show it instead:

```
  def terminate(): Future[Terminated]
```

Terminating is a one liner kind of now.

`whenTerminated` is more for when different parts of the app want to do stuff on shutdown. You can show it as well here :) Add a note to NOT use an `context.dispatcher` when `.map()` in on the Future, since that dispatcher may have been shut down already, since we're shutting down, right :)

// I see this note is missing from the scaladoc on terminate / whenTerminated - maybe add it there as well?

https://github.com/akka/akka/blob/master/akka-actor/src/main/scala/akka/actor/ActorSystem.scala#L431